File: python/secret-handshake/secret_handshake.py
def commands(binary_str):
    """
    Convert a binary string to a list of secret handshake actions.

    Args:
        binary_str (str): A binary string representing a sequence of secret handshake actions.
            The rightmost digit corresponds to the first action, and the leftmost digit to the last action.

    Returns:
        list: A list of secret handshake actions.

    Examples:
        commands("11010")
        # Output: ['jump', 'double blink']

    Explanation:
        The input binary_str represents the actions 'jump' and 'double blink' in this order.

    """
    # Reverse the binary string to start with the rightmost digit.
    rvs = binary_str[::-1]

    result = []
    
    # Define a dictionary to map binary digits to actions.
    adict = {
        0: "wink",
        1: "double blink",
        2: "close your eyes",
        3: "jump",
    }

    # Iterate through the reversed binary string to extract actions.
    for i in range(len(rvs) - 1):
        if rvs[i] == "1":
            result.append(adict[i])

    # Check the leftmost digit to determine if actions should be reversed.
    if binary_str[0] == "1":
        result = result[::-1]

    return result

File: javascript/secret-handshake/secret-handshake.js
export const commands = (input) => {
  // Define the actions corresponding to each digit
  const actions = ['wink', 'double blink', 'close your eyes', 'jump'];
  
  // Convert the input number to a binary string and pad it with zeros on the left to ensure ti has 5 digits
  const binary = input.toString(2).padStart(5, '0');

  // Split the binary string into an array of digits, reverse the array, and map each digit to its corresponding action
  const sequence = binary.split('').reverse().map((digit, index) => {
    // If the digit is 1, map it to its corresponding action
    if (digit === '1') {
      return actions[index];
    }
  // Remove any undefined values from the array, which occur when the digit is 0
  }).filter(Boolean);

  // If the 5th digit from the right is 1, reverse the sequence
  if (binary[0] === '1') {
    return sequence.reverse();
  }

  // Return the sequence
  return sequence;
};
